source/dataman/database/AMDatabase.cpp

```diff
@@ -140,7 +140,7 @@ int AMDatabase::insertOrUpdate(int id, const QString& table, const QStringList&
 
 	if(!db.isOpen()) {
 		AMErrorMon::report(AMErrorReport(this, AMErrorReport::Alert, -2, "Could not save to database. (Database is not open.)"));
-		return false;
+		return 0;
```

Comment:
The signature of this function is to return an integer value. I bet this was typo long time ago since we always return 0 except this one when sth is wrong.